src/Polysemy/Internal/Scoped.hs

```diff
@@ -21,30 +22,138 @@ import Polysemy.Internal.Combinators (transform)
 --
 -- For a longer exposition, see <https://www.tweag.io/blog/2022-01-05-polysemy-scoped/>.
 --
--- Resource creation is performed by the function passed to 'Polysemy.Scoped.runScoped'.
+-- Resource allocation is performed by a function passed to 'Polysemy.Scoped.interpretScoped'.
 --
 -- The constructors are not intended to be used directly; the smart constructor 'scoped' is used like a local
 -- interpreter for @effect@.
-data Scoped (resource :: Type) (effect :: Effect) :: Effect where
-  Run :: ∀ resource effect m a . resource -> effect m a -> Scoped resource effect m a
-  InScope :: ∀ resource effect m a . (resource -> m a) -> Scoped resource effect m a
+-- 'scoped' takes an argument of type @param@, which will be passed through to the interpreter, to be used by the
+-- resource allocation function.
+--
+-- As an example, imagine an effect for writing lines to a file:
+--
+-- > data Write :: Effect where Write :: Text -> Write m ()
+-- > makeSem ''Write
+--
+-- If we now have the following requirements:
+--
+-- 1. The file should be opened and closed right before and after the part of the program in which we write lines
+-- 2. The file name should be specifiable at the point in the program where writing begins
+-- 3. We don't want to commit to IO, lines should be stored in memory when running tests
```

Comment:
I think this example is great, it seems minimal and concretely shows under when `Scoped` is useful. Personally I would add something along the lines of `(notice how these requirements are very similar to what we have in regards to transactions and a connection in databases)`, but maybe that use case is not that prevalent.

Reply:
Although I guess in that case usually you reuse the same connection, hence `Scoped_` would be closer.

Reply:
well the parameter could be something else, like a flag indicating rollback behaviour?

Reply:
Right.